WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/Player.py

```diff
@@ -26,11 +27,11 @@ def __init__(self, id: int, deck: Deck):
     def get_hp(self) -> int:
         return self.statistics.hp
 
-    def use_card(self) -> list[Effect]:
+    def use_card(self) -> tuple[BattleCard, list[Effect]]:
         """
         Uses card which is first in deck to use and then places that card at the end of the deck.
-        @return: List of effects of proper card to be executed by battle simulation.
+        @return: Proper card and it's list of effects to be executed by battle simulation .
         """
 
         card = self.deck.get_card()
-        return card.use()
+        return card, card.use()
```

Comment:
Dlaczego zwracamy osobno listę efektów? Przecież `BattleCard` ma w sobie efekty. `card.use()` zwraca `self.effects`

Reply:
Zwracanie listy efektów zrobiliśmy już wcześniej, ja dołożyłem też zwracanie karty, bo potrzebna jest też informacja jaka karta była użyta

Reply:
Według mnie tak jest dobrze, inaczej to `Coordinator` musiałby najpierw od gracza wziąć kartę, którą ten ma zamiar użyć, a potem dopiero wywołać `use_card` na graczu. Teraz gracz używa kartę, i od razu mamy zwrócone jaką kartę użył i jakie efekty `Coordinator` ma aktywować po kolei.